### PR TITLE
ADD ViewStateString

### PR DESCRIPTION
**Pull request by Frank1234**
ViewStateStrings enabled you to move String logic from the View to the ViewModel, especially plurals
or replacement parameter logic.

### CODE REVIEW

File: core/ui/src/main/kotlin/nl/q42/template/ui/presentation/ViewStateString.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    data class PluralRes(
+        @PluralsRes val pluralRes: Int,
+        val count: Int,
+        val formatArgs: List<Any> = listOf(count)
+    ) : ViewStateString() {
+        // Allow constructing ViewStateString.PluralRes with varargs instead of passing a list
+        @Suppress("unused")
+        constructor(pluralRes: Int, count: Int, vararg formatArgs: Any) : this(
```
Comment (sebaslogen) [inline core/ui/src/main/kotlin/nl/q42/template/ui/presentation/ViewStateString.kt:31]: Is the @PluralsRes  annotation not needed/valid here?

File: core/ui/src/main/kotlin/nl/q42/template/ui/compose/ViewStateStringResolver.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.res.pluralStringResource
+import androidx.compose.ui.res.stringResource
+import nl.q42.template.ui.presentation.ViewStateString
+
+/**
+ * Resolve a ViewStateString to a string from a Compose context.
+ */
+@Composable
+fun ViewStateString.resolve(): String {
```
Comment (sebaslogen) [inline core/ui/src/main/kotlin/nl/q42/template/ui/compose/ViewStateStringResolver.kt:15]: I would have personally named this function toString() to keep it familiar but this name is good enough.

**Review by sebaslogen — APPROVED** [review]
I found just one typo twice.
LGTM

File: core/ui/src/main/kotlin/nl/q42/template/ui/compose/ViewStateStringResolver.kt (hunk 2 above)
Comment (Frank1234) [reply core/ui/src/main/kotlin/nl/q42/template/ui/compose/ViewStateStringResolver.kt:15]: toString feels improper. It usually prints a description.

any other options? get() ?

Comment (sebaslogen) [reply core/ui/src/main/kotlin/nl/q42/template/ui/compose/ViewStateStringResolver.kt:13]: Yes, I like "get", short and sweet

**Review by Techwolf12 — DISMISSED** [review]
Code LGTM! Unit test still needs a fix

File: core/ui/src/main/kotlin/nl/q42/template/ui/compose/ViewStateStringResolver.kt (hunk 2 above)
Comment (Techwolf12) [reply core/ui/src/main/kotlin/nl/q42/template/ui/compose/ViewStateStringResolver.kt:13]: +1 for get(), clear in what to expect

**Pull request merged**